docs/concepts/sync-vs-async.rst

```diff
@@ -0,0 +1,117 @@
+===========================
+Sync vs Async Applications
+===========================
+
+TL;DR
```

Comment:
Link to the `arun`, `aiterate`, etc... methods. Good to tie it back into the actions.